src/tbb/allocator.cpp

```diff
@@ -115,7 +116,9 @@ static const dynamic_link_descriptor MallocLinkTable[] = {
     If that allocator is not found, it links to malloc and free. */
 void initialize_handler_pointers() {
     __TBB_ASSERT(allocate_handler == &initialize_allocate_handler, NULL);
-    bool success = dynamic_link(MALLOCLIB_NAME, MallocLinkTable, 4);
+    bool success = false;
+    if (!GetBoolEnvironmentVariable("TBB_USE_MALLOC"))
+        success = dynamic_link(MALLOCLIB_NAME, MallocLinkTable, 4);
```

Comment:
Have you considered an approach to extend `global_control` functionality? Usually, we try to avoid configuration via environment variables.

Reply:
Yes, the rationale and discussion about `global_control` is in the previous version of this PR: https://github.com/oneapi-src/oneTBB/pull/149. The short version is that we want to ensure that tbbmalloc is not enabled if TBB is used somewhere during global initialization of the program, and we want to propagate that setting to child processes.